components/LanguageSwitcher.vue

```diff
@@ -0,0 +1,225 @@
+<template>
+  <div
+    v-if="showLanguageDropdown && type === 'with-dropdown'"
+    class="language-switcher"
+  >
+    <button
+      class="language-switcher__button"
+      :aria-label="$t('language-switcher-label')"
+      aria-haspopup="true"
+      :aria-expanded="showMenu.toString()"
+      @click="toggleMenu()"
+      @click.stop.prevent
+    ></button>
+    <ul v-if="showMenu" class="language-switcher__dropdown-menu">
+      <li v-for="option in availableLocales" :key="option.key">
+        <span
+          v-if="option === currentLanguage"
+          class="language-switcher__lang--active"
+          >{{ option.name }}</span
+        >
+        <span v-else class="language-switcher__lang">
+          <a href="javascript:void(0)" @click="update(option)">
+            {{ option.name }}
+          </a>
+        </span>
+      </li>
+    </ul>
+    <span class="separator"></span>
+  </div>
+  <div v-else-if="showLanguageDropdown && type === 'only-text'">
+    <ul>
+      <li v-for="option in availableLocales" :key="option.key">
+        <span
+          v-if="option !== currentLanguage"
+          class="language-switcher-text__lang"
+        >
+          <a href="javascript:void(0)" @click="update(option)">
+            {{ option.name }}
+          </a>
+        </span>
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'LanguageSwitcher',
+  props: {
+    type: {
+      type: String,
+      default: 'with-dropdown',
+    },
+  },
+  data() {
+    const allLocales = this.$i18n.locales.map((locale) => {
+      return { name: this.$t(locale.code), lang: locale.code }
+    })
+    const availableLocales = allLocales.filter(
+      (locale) => locale.lang !== 'fr-fr'
+    )
+    return {
+      showMenu: false,
+      availableLocales,
+    }
+  },
+  computed: {
+    showLanguageDropdown() {
+      return (
+        process.env.isPixSite &&
+        this.$config.languageSwitchEnabled &&
+        this.$i18n.locale !== 'fr-fr'
+      )
+    },
+    currentLocale() {
+      return this.$i18n.locale || this.$i18n.defaultLocale
+    },
+    currentLanguage() {
+      return this.availableLocales.find(
+        (locale) => locale.lang === this.currentLocale
+      )
+    },
+  },
+  mounted() {
+    const page = document.getElementsByTagName('body')[0]
+    const menu = document.getElementsByClassName('language-switcher')[0]
+    page.addEventListener('click', this.hideMenu)
+    menu.addEventListener('keydown', this.hideMenu)
```

Comment:
```suggestion
    const page = document.getElementsByTagName('body')[0]
    page.addEventListener('click', this.hideMenu)
    const menu = document.getElementsByClassName('language-switcher')[0]
    menu.addEventListener('keydown', this.hideMenu)
```

Et sortir 2 méthodes : listenEscape et listenClickOutside
Voir pour utiliser this.$el

Reply:
Utiliser `@keydown.esc.exact ` ?